### PR TITLE
Clean up ClusterProperty

### PR DESCRIPTION
**Pull request by mmedenjak**
- arrange properties in groups
- add javadoc from reference manual

No changes in behaviour.

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2393/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2393/Hazelcast-pr-builder-2393.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There are 2 errors reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1: File length is 1,574 lines (max allowed is 1,500). [FileLength]
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:103:43: Name 'PREFER_IPv4_STACK' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2394/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2394/Hazelcast-pr-builder-2394.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_2/hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1: File length is 1,575 lines (max allowed is 1,500). [FileLength]
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2395/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2395/Hazelcast-pr-builder-2395.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   MembershipFailureTest.test_twoSlavesDisconnectedFromOneSlave_when_clusterState_NO_MIGRATON:829->test_twoSlavesDisconnectedFromOneSlave:855 cluster members: [Member [127.0.0.1]:5701 - 94b4f029-ebc1-4a34-9c47-81bb4e11acfc this, Member [127.0.0.1]:5703 - b034af16-048c-4d2f-b0c8-f34dea1bb828, Member [127.0.0.1]:5704 - 0230d1da-d0c4-431b-a825-ee0ea26c0925]
[INFO] 
[ERROR] Tests run: 3878, Failures: 1, Errors: 0, Skipped: 14
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Review by ahmetmircik — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java [hunk 1]
```diff
@@ -53,170 +53,177 @@
...
+     * Size is in kilobytes.
+     * <p>
+     * The default is -1 (same as receive buffer size for connections opened by members).
      */
-    public static final HazelcastProperty EVENT_THREAD_COUNT
-            = new HazelcastProperty("hazelcast.event.thread.count", 5);
+    public static final HazelcastProperty SOCKET_CLIENT_SEND_BUFFER_SIZE
+            = new HazelcastProperty("hazelcast.socket.client.send.buffer.size", -1);
+
+    public static final HazelcastProperty SOCKET_CLIENT_BUFFER_DIRECT
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:186]: Could you also add the JavaDoc?

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java [hunk 2]
```diff
@@ -483,417 +493,444 @@ private int getWhenNoSSLDetected() {
...
             = new HazelcastProperty("hazelcast.member.list.publish.interval.seconds", 60, SECONDS);
 
+    /**
+     * Time after which the member assumes the client is dead and closes its
+     * connections to the client.
+     */
     public static final HazelcastProperty CLIENT_HEARTBEAT_TIMEOUT_SECONDS
             = new HazelcastProperty("hazelcast.client.max.no.heartbeat.seconds", 300, SECONDS);
 
     public static final HazelcastProperty CLUSTER_SHUTDOWN_TIMEOUT_SECONDS
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:580]: Could you also add the JavaDoc?

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java [hunk 3]
```diff
@@ -483,417 +493,444 @@ private int getWhenNoSSLDetected() {
...
     /**
-     * Defines the maximum number of Near Cache invalidation events to be drained and sent to the event Near Cache in a batch.
+     * Enable JMX agent.
      */
-    public static final HazelcastProperty MAP_INVALIDATION_MESSAGE_BATCH_SIZE
-            = new HazelcastProperty("hazelcast.map.invalidation.batch.size", 100);
+    public static final HazelcastProperty ENABLE_JMX
+            = new HazelcastProperty("hazelcast.jmx", false);
+
+    public static final HazelcastProperty JMX_UPDATE_INTERVAL_SECONDS
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:758]: Could you also add the JavaDoc?

**Review by kwart — APPROVED** [review]
It would be nice to have documented all the cluster properties. Otherwise LGTM.

**Comment by mmedenjak** [comment]
The only ones that didn't have javadoc are the ones which didn't have documentation in the reference manual :)

I'll see if I can figure out what some property is doing and add javadoc for that as well.

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java (hunk 2 above)
Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:590]: Added. I used my power of reasoning and tried to write a nice, fresh javadoc :) Maybe I can add the same now to the reference manual 🤔

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java (hunk 1 above)
Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:190]: Added.

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java (hunk 3 above)
Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:774]: Added.

**Comment by mmedenjak** [comment]
Thanks for the reviews!

**Pull request merged**